crypto/lhash/lhash.c

```diff
@@ -156,7 +156,26 @@ void *OPENSSL_LH_retrieve(OPENSSL_LHASH *lh, const void *data)
     unsigned long hash;
     OPENSSL_LH_NODE **rn;
     void *ret;
+#ifdef HAVE_ATOMICS
+    CRYPTO_REF_TYPE junk = 0;
+# ifdef __GNUC__
+    int nil = 0;
 
+    __atomic_store(&lh->error, &nil, __ATOMIC_RELEASE);
+# else
+    atomic_store(&lh->error, 0);
+# endif
```

Comment:
We have `CRYPTO_atomic_write`, I would see that used rather then adding another local preprocessor soup. If you think `CRYPTO_atomic_write` needs fixing, then do that.

Reply:
This is effectively answered by mere existence of this request as alternative to #6754. CRYPTO_atomic_write requires instance of lock, and we don't have one. Nor do we want one, because caller has one that customarily protects bigger structure. As for fixing CRYPTO_atomic_write, I for one would rather go as far as *removing* it...